### ISSUE TITLE
Not Compiling on Windows 

### ISSUE THREAD

**Issue by mkhraijah** [kahypar/kahypar#135]
I noticed that there is an issue with the package not compiling on Windows systems. Is there a plan to fix the issue? 

Thanks,

**Comment by SebastianSchlag** [comment]
Hi @mkhraijah, 
We haven't tried to compile KaHyPar on windows for a while. There was a point where we got it to work though. I think at the moment CI mainly  fails because of some setup/dependency issues. At the moment, we don't really plan to support Windows explicitly as it can be run on Windows machines through the [Windows Subsystem for Linux](https://learn.microsoft.com/en-us/windows/wsl/install). Would that work for your purposes?

**Comment by mkhraijah** [comment]
Thanks for your response! I'm actually using Linux to run the package and it works very well. I'm using kahypar on a Julia package that we are developing and I received an issue from a user about not being able to compile the package on Windows because of kahypar.

**Comment by SebastianSchlag** [comment]
Could you ask your user for the error message that they got? Maybe it is not even that difficult to get Windows builds up and running again.

**Comment by mkhraijah** [comment]
Please see the below photo for the error message when precompiling the package. 

![error2](https://user-images.githubusercontent.com/43002862/217713273-30b9522a-861a-4555-8f4e-2eda61cacb45.png)

**Comment by SebastianSchlag** [comment]
This seems to be an issue with the Julia interface, right? Could you please open up an issue over at @jalving's [KaHyPar.jl](https://github.com/kahypar/KaHyPar.jl) project?

**Comment by SebastianSchlag** [comment]
Never mind, this actually seems to be related to https://github.com/kahypar/KaHyPar.jl/issues/26. If I find some time, I'll take a look to find out what it takes to get this up and running on Windows.

**Comment by Diggsey** [comment]
FWIW, I just build the project on windows and the only issue was the tests. After disabling the test projects in the MSVC solution, it all build just fine.

**Comment by SebastianSchlag** [comment]
Thank you very much for letting us know @Diggsey. Do you have an idea what went wrong with the tests? Could you provide us with the error logs?

**Comment by Diggsey** [comment]
These are the errors from one of the test builds (the others are similar):
```
3>------ Build started: Project: binary_heap_test, Configuration: Release x64 ------
3>cl : command line  warning D9025: overriding '/W4' with '/w'
3>binary_heap_test.cc
3>kahypar/external_tools/googletest/googletest/include\gtest/internal/gtest-port.h(1002,7): error C2039: 'tr1': is not a member of 'std'
3>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.33.31629\include\vector(24): message : see declaration of 'std'
3>kahypar/external_tools/googletest/googletest/include\gtest/internal/gtest-port.h(1002,7): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/internal/gtest-port.h(1003,7): error C2039: 'tr1': is not a member of 'std'
3>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.33.31629\include\vector(24): message : see declaration of 'std'
3>kahypar/external_tools/googletest/googletest/include\gtest/internal/gtest-port.h(1003,7): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/internal/gtest-port.h(1004,7): error C2039: 'tr1': is not a member of 'std'
3>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.33.31629\include\vector(24): message : see declaration of 'std'
3>kahypar/external_tools/googletest/googletest/include\gtest/internal/gtest-port.h(1004,7): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/internal/gtest-port.h(1005,7): error C2039: 'tr1': is not a member of 'std'
3>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.33.31629\include\vector(24): message : see declaration of 'std'
3>kahypar/external_tools/googletest/googletest/include\gtest/internal/gtest-port.h(1005,7): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/internal/gtest-port.h(1006,7): error C2039: 'tr1': is not a member of 'std'
3>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.33.31629\include\vector(24): message : see declaration of 'std'
3>kahypar/external_tools/googletest/googletest/include\gtest/internal/gtest-port.h(1006,7): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(600,34): error C2039: 'tr1': is not a member of 'std'
3>C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.33.31629\include\set(23): message : see declaration of 'std'
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(600,34): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(605,27): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(610,27): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(615,27): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(620,27): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(625,27): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(632,27): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(639,27): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(646,27): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(653,27): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(661,18): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(878,33): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(878): message : see reference to class template instantiation 'testing::internal::TuplePolicy<TupleT>::tuple_element<I>' being compiled
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(886): message : see reference to class template instantiation 'testing::internal::TuplePolicy<TupleT>' being compiled
3>kahypar/external_tools/googletest/googletest/include\gtest/gtest-printers.h(882,29): error C3083: 'tr1': the symbol to the left of a '::' must be a type
3>Done building project "binary_heap_test.vcxproj" -- FAILED.
```

It looks like this issue - https://stackoverflow.com/questions/42847103/stdtr1-with-visual-studio-2017 - albeit with VS2022 rather than 2017. Looks like the feature detection in Google's GTest is at fault.

**Comment by SebastianSchlag** [comment]
Thanks a lot @Diggsey. One comment in the stackoverflow post you shared mentioned that this should be fixed with googletest 1.8.1. I updated KaHyPar's submodule in https://github.com/kahypar/kahypar/pull/173 to pull in the latest `googletest` version.
Would you mind trying out that branch to check whether this resolves the issue?
